util/helm/cmd.go

```diff
@@ -196,14 +197,18 @@ func (c *Cmd) template(chart string, opts *TemplateOpts) (string, error) {
 		args = append(args, "--set-string", key+"="+cleanSetParameters(val))
 	}
 	for _, val := range opts.Values {
-		absWorkDir, err := filepath.Abs(c.WorkDir)
+		absBaseDir, err := filepath.Abs(opts.BaseDirectoryPath)
 		if err != nil {
 			return "", err
 		}
 		if !filepath.IsAbs(val) {
+			absWorkDir, err := filepath.Abs(c.WorkDir)
+			if err != nil {
+				return "", err
+			}
 			val = filepath.Join(absWorkDir, val)
 		}
-		cleanVal, err := security.EnforceToCurrentRoot(absWorkDir, val)
+		cleanVal, err := security.EnforceToCurrentRoot(absBaseDir, val)
```

Comment:
I'm not sure security checks should be the responsibility to `cmd.go`, not it must know about the repo's root, and this seems like a smell to me. Can we move these higher up? I.e. check the paths when creating `templateOpts`?

Reply:
Sure, we can move this higher up. My logic here was running the directory enforcer as close to the command as possible to avoid unchecked paths to sneak files in. But I can see how this could be an issue.